lib/user.js

```diff
@@ -43,6 +45,71 @@ class UserModel extends BaseModel {
             [this.token, this[password], iron.defaults]);
     }
 
+    /** Fetch a user's tokens
+    /* @property tokens
+    /* @return {Promise}
+    */
+    get tokens() {
+        const listConfig = {
+            params: {
+                user: this.id
+            },
+            paginate: {
+                count: PAGINATE_COUNT,
+                page: PAGINATE_PAGE
+            }
+        };
+
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const TokenFactory = require('./tokenFactory');
+        /* eslint-enable global-require */
+        const factory = TokenFactory.getInstance();
+
+        const tokens = factory.list(listConfig);
+
+        // ES6 has weird getters and setters in classes,
+        // so we redefine the pipeline property here to resolve to the
+        // resulting promise and not try to recreate the factory, etc.
+        Object.defineProperty(this, 'tokens', {
+            enumerable: true,
+            value: tokens
+        });
+
+        return tokens;
+    }
+
+    /**
+     * Test if a token is valid and belongs to the user
+     * @method validateToken
+     * @param  {String}  uuid       UUID of token to validate
+     * @return {Promise}
+     */
+    validateToken(uuid) {
+        return this.tokens.then((tokens) => {
+            const match = tokens.find(token => token.uuid === uuid);
+
+            if (!match) {
+                throw new Error('Token has been revoked.');
```

Comment:
In the API layer, to authenticate first we check that the token is valid with the public key. If it fails there, it doesn't have to bother checking to see if the token is still valid or not because it already knows it isn't. So it should only get to here if it would be a valid token except for the fact that it's been revoked.